android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -932,4 +942,57 @@ public String dataChannelStateString(DataChannel.State dataChannelState) {
         }
         return null;
     }
+
+    class DataChannelObserver implements DataChannel.Observer {
```

Comment:
It would be good to have the code better separated, but currently all of listeners/interface impls reside in WebRTCModule. I would say it's fine, because this is just a glue layer and would not expect any complicated logic there. But I've been thinking about extracting "GUM" constraints parsing into utility class to get rid of specialised code from WebRTCModule.